tests/test_features_feature_collection.py

```diff
@@ -24,11 +24,51 @@
 )
 from tsflex.utils.data import flatten
 
-from .utils import dummy_data
+from .utils import dummy_data, dummy_group_data  # noqa: F401
 
 ## FeatureCollection
 
 
+def test_single_series_group_feature_collection(dummy_group_data):
+    fd = FeatureDescriptor(
+        function=np.sum,
+        series_name="count",
+    )
+
+    fc = FeatureCollection(feature_descriptors=fd)
+
+    assert fc.get_required_series() == ["count"]
+    assert fc.get_nb_output_features() == 1
+    res_list = fc.calculate(dummy_group_data, group_by="country", return_df=False)
+    res_df = fc.calculate(dummy_group_data, group_by="country", return_df=True)
+
+    assert isinstance(res_list, list)
+    assert isinstance(res_df, pd.DataFrame)
+    assert np.all(dummy_group_data["country"].unique() == res_df.index.unique())
+
+    for idx, frame in enumerate(res_list):
+        assert_frame_equal(frame, res_df.iloc[(idx) : (idx + 1)])
+
+    data_counts = dummy_group_data.groupby("country")["count"].sum()
+
+    for idx in data_counts.index:
+        assert res_df.loc[idx, "count__sum__w=manual"] == data_counts.loc[idx]
+
+
+def test_single_series_group_feature_non_existent_group_by(dummy_group_data):
+    fd = FeatureDescriptor(
+        function=np.sum,
+        series_name="count",
+    )
+
+    fc = FeatureCollection(feature_descriptors=fd)
+
+    assert fc.get_required_series() == ["count"]
+    assert fc.get_nb_output_features() == 1
+    with pytest.raises(Exception):
+        fc.calculate(dummy_group_data, group_by="nonexistent", return_df=False)
+
```

Comment:
I believe we are still missing some tests here. 

- What is the expected behavior when the input is a series list that does not have an equal length?
- How are Nan group column values handled?
- What should occur when non-aligned indices are-used (e.g. the grouping column has time-ranges) (which intersect with the value-column)

```python
# create the feature collection
fc = FeatureCollection(FeatureDescriptor(function=np.sum, series_name="count"))

# create the dummy data
s_group = pd.Series(
    index=pd.date_range('2018-01-01', freq='10m', periods=30), name='user_id',
    # data=['a'] * 10 + ['b'] * 2 + ['c'] + ['d'] * 2 + ['e'] * 15 
    # data=['a'] * 10 + ['b'] * 2 + ['c'] + ['d'] * 2 + [None] + ['e'] * 14
    data=['a'] * 10 + ['b'] * 2 + ['c'] + ['d'] * 2 + [None] * 3+ ['e'] * 12 
)
s_val = pd.Series(index=pd.date_range('2018-01-01', freq='3m', periods=100), data=np.arange(100), name='count')
res_list = fc.calculate([s_group, s_val], group_by="user_id", return_df=True)
res_list
```

Reply:
As I've read below in the function, the groups will be only created on the intersection of the index-vals between the grouping and value columns.